authorization/v2/authorizer.go

```diff
@@ -0,0 +1,94 @@
+package v2
+
+import (
+	"github.com/dgrijalva/jwt-go"
+	"github.com/pkg/errors"
+	"github.com/unicsmcr/hs_auth/environment"
+	"github.com/unicsmcr/hs_auth/utils"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+const unknownTokenTypeErrTemplate = "'%s' is not a valid token type"
+
+// Authorizer provides an interface for creating auth tokens and checking their permissions
+type Authorizer interface {
+	// CreateUserToken creates a token for the given user.
+	// Setting expirationDate to 0 will create a token that does not expire.
+	CreateUserToken(userId primitive.ObjectID, expirationDate int64) (string, error)
+	// CreateServiceToken creates a token for the given owner with the provided permissions.
+	// Setting expirationDate to 0 will create a token that does not expire.
+	CreateServiceToken(owner string, allowedResources []UniformResourceIdentifier, timeToLive int64) (string, error)
+	// GetAuthorizedResources returns what resources from urisToCheck the given token can access.
+	// Will return ErrInvalidToken if the provided token is invalid.
+	GetAuthorizedResources(token string, urisToCheck []UniformResourceIdentifier) ([]UniformResourceIdentifier, error)
+}
+
+
+func NewAuthorizer(provider utils.TimeProvider, env *environment.Env) Authorizer {
+	return &authorizer{
+		timeProvider: provider,
+		env:          env,
+	}
+}
+
+type authorizer struct {
+	timeProvider utils.TimeProvider
+	env          *environment.Env
+}
+
+func (a *authorizer) CreateUserToken(userId primitive.ObjectID, expirationDate int64) (string, error) {
+	timestamp := a.timeProvider.Now().Unix()
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, TokenClaims{
```

Comment:
Since the `jwt.SigningMethodHS256` is used in both the creation of the user token and the service token. Would it be possible to extract that to a const variable that can be reused.